src/translate_ligo.coffee

```diff
@@ -376,41 +380,49 @@ class @Gen_context
     t.structs_default_list = @structs_default_list
     t.enum_list = @enum_list
     t.contract = @contract
+    t.files = @files
+    t.keep_dir_structure = @keep_dir_structure
     t
 
 last_bracket_state = false
 walk = (root, ctx)->
+  main_file = ""
   last_bracket_state = false
   switch root.constructor.name
     when "Scope"
       switch root.original_node_type
         when "SourceUnit"
-          jl = []
+          jls = {}
+          jls[main_file] = []
           for v in root.list
             code = walk v, ctx
+            path = v.file if ctx.keep_dir_structure
```

Comment:
second iteration will reuse path from first iteration
Is that intentional?

Reply:
How do you know? No it wasn't intentional